enumerator/fake_enumerable.rb

```diff
@@ -0,0 +1,18 @@
+module FakeEnumerable
```

Comment:
Great job on that, buy one more thing i want you to do.

https://blog.arkency.com/2014/01/ruby-to-enum-for-enumerator/

Basically, write custom iterator using ```#Enumerator.new``` and ```#to_enum & #enum_for``` so ypu would really have Enumerator instance